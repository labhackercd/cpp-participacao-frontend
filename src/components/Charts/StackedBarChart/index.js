import React from 'react';
//import Plot from 'react-plotly.js';
import Chart from "react-google-charts";


export default function StackedBarChart(props) {
  return (
      <Chart
        width={'100%'}
        height={"115px"}
        chartType="BarChart"
        loader={<div>Carregando Gr√°fico</div>}
        data={props.data}
        options={props.options}
      />
   );
}