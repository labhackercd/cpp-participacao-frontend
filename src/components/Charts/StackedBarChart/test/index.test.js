import React from 'react';
import ReactDOM from 'react-dom';
import {mount} from "enzyme";
import StackedBarChart from '../index';

const stackedBarChartData = {
    values:[
        ['Ferramenta', 'AudiÃªncias Interativas', 'Wikilegis', 'Enquetes', 'Pauta Participativa'],
        ['',300, 800, 231, 545],
    ],
    options:{
        legend: { position: 'top', maxLines: 3,textStyle: {color: 'white'} },
        isStacked: 'relative',
        colors: ['#FFC107', '#76D275', '#6EC6FF', '#FF6F60'],
        bar: { groupWidth: '30%'},
        hAxis: {minValue: 0,gridlines: {color: 'transparent'}, textPosition: 'none', },
        backgroundColor: '#000000',
    }
};

test("Snapshot should not have changes", () => {
    const component = mount(<StackedBarChart data={stackedBarChartData.values} options={stackedBarChartData.options}></StackedBarChart>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

it("Mounts the chart with the data", () => {
    const div = document.createElement("div")
    ReactDOM.render(<StackedBarChart data={stackedBarChartData.values} options={stackedBarChartData.options}></StackedBarChart>, div)
});
