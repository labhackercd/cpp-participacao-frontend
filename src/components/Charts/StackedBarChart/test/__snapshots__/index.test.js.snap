// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should not have changes 1`] = `
<StackedBarChart
  data={
    Array [
      Array [
        "Ferramenta",
        "Audiências Interativas",
        "Wikilegis",
        "Enquetes",
        "Pauta Participativa",
      ],
      Array [
        "",
        300,
        800,
        231,
        545,
      ],
    ]
  }
  options={
    Object {
      "backgroundColor": "#000000",
      "bar": Object {
        "groupWidth": "30%",
      },
      "colors": Array [
        "#FFC107",
        "#76D275",
        "#6EC6FF",
        "#FF6F60",
      ],
      "hAxis": Object {
        "gridlines": Object {
          "color": "transparent",
        },
        "minValue": 0,
        "textPosition": "none",
      },
      "isStacked": "relative",
      "legend": Object {
        "maxLines": 3,
        "position": "top",
        "textStyle": Object {
          "color": "white",
        },
      },
    }
  }
>
  <Chart
    chartActions={null}
    chartEvents={null}
    chartType="BarChart"
    chartWrapperParams={Object {}}
    className=""
    columns={null}
    controls={null}
    data={
      Array [
        Array [
          "Ferramenta",
          "Audiências Interativas",
          "Wikilegis",
          "Enquetes",
          "Pauta Participativa",
        ],
        Array [
          "",
          300,
          800,
          231,
          545,
        ],
      ]
    }
    diffdata={null}
    formatters={null}
    getChartEditor={null}
    getChartWrapper={[Function]}
    graphID={null}
    graph_id={null}
    height="115px"
    legendToggle={false}
    legend_toggle={false}
    loader={
      <div>
        Carregando Gráfico
      </div>
    }
    options={
      Object {
        "backgroundColor": "#000000",
        "bar": Object {
          "groupWidth": "30%",
        },
        "colors": Array [
          "#FFC107",
          "#76D275",
          "#6EC6FF",
          "#FF6F60",
        ],
        "hAxis": Object {
          "gridlines": Object {
            "color": "transparent",
          },
          "minValue": 0,
          "textPosition": "none",
        },
        "isStacked": "relative",
        "legend": Object {
          "maxLines": 3,
          "position": "top",
          "textStyle": Object {
            "color": "white",
          },
        },
      }
    }
    render={null}
    rootProps={Object {}}
    rows={null}
    spreadSheetQueryParameters={
      Object {
        "gid": 1,
        "headers": 1,
      }
    }
    spreadSheetUrl={null}
    style={Object {}}
    toolbarID={null}
    toolbarItems={null}
    width="100%"
  >
    <ContextProvider
      value={
        Object {
          "chartActions": null,
          "chartEvents": null,
          "chartType": "BarChart",
          "chartWrapperParams": Object {},
          "className": "",
          "columns": null,
          "controls": null,
          "data": Array [
            Array [
              "Ferramenta",
              "Audiências Interativas",
              "Wikilegis",
              "Enquetes",
              "Pauta Participativa",
            ],
            Array [
              "",
              300,
              800,
              231,
              545,
            ],
          ],
          "diffdata": null,
          "formatters": null,
          "getChartEditor": null,
          "getChartWrapper": [Function],
          "graphID": null,
          "graph_id": null,
          "height": "115px",
          "legendToggle": false,
          "legend_toggle": false,
          "loader": <div>
            Carregando Gráfico
          </div>,
          "options": Object {
            "backgroundColor": "#000000",
            "bar": Object {
              "groupWidth": "30%",
            },
            "colors": Array [
              "#FFC107",
              "#76D275",
              "#6EC6FF",
              "#FF6F60",
            ],
            "hAxis": Object {
              "gridlines": Object {
                "color": "transparent",
              },
              "minValue": 0,
              "textPosition": "none",
            },
            "isStacked": "relative",
            "legend": Object {
              "maxLines": 3,
              "position": "top",
              "textStyle": Object {
                "color": "white",
              },
            },
          },
          "render": null,
          "rootProps": Object {},
          "rows": null,
          "spreadSheetQueryParameters": Object {
            "gid": 1,
            "headers": 1,
          },
          "spreadSheetUrl": null,
          "style": Object {},
          "toolbarID": null,
          "toolbarItems": null,
          "width": "100%",
        }
      }
    >
      <div>
        Carregando Gráfico
      </div>
      <GoogleChartLoader
        onError={[Function]}
        onLoad={[Function]}
      >
        <Script
          attributes={Object {}}
          onCreate={[Function]}
          onError={[Function]}
          onLoad={[Function]}
          url="https://www.gstatic.com/charts/loader.js"
        />
      </GoogleChartLoader>
    </ContextProvider>
  </Chart>
</StackedBarChart>
`;
